### PR TITLE
Organization settings & User tab

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: src/app/shared/input/filter-box/filter-box.component.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  @Input() public value: string;
+
+  @Input() public placeholder: string;
+
+  @Output() public textChange: EventEmitter<string> = new EventEmitter();
+
+  @Output() public textSubmit: EventEmitter<string> = new EventEmitter();
+
+  public mPlaceholder: string;
+  public mCurrentValue: string;
```
Comment (livthomas) [inline src/app/shared/input/filter-box/filter-box.component.ts]: I'm just curious, why `m` prefix?

File: src/app/shared/utils/validator.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export class Validator {
```
Comment (livthomas) [inline src/app/shared/utils/validator.ts]: We already have validators in the core module. So it should be moved there and rename accordingly.

File: src/app/view/view-controls/share-dialog/share-dialog.component.ts [hunk 3]
```diff
@@ -17,35 +17,54 @@
...
   public constructor(private notificationService: NotificationService,
-                     private userService: UserService) {
+                     private store: Store<AppState>) {
   }
 
   public ngOnInit() {
-    this.userService.getUsers().subscribe((users: User[]) => {
-      this.users = users;
-    });
+    this.subscribeData();
```
Comment (livthomas) [inline src/app/view/view-controls/share-dialog/share-dialog.component.ts]: +1 on making this reactive.

File: src/app/workspace/organization/organization-settings.component.html [hunk 4]
```diff
@@ -1,98 +1,103 @@
...
-          <icon-picker class="dropdown-item" [(icon)]="organization.icon"
-                       (selectedChange)="updateOrganization()"></icon-picker>
-        </div>
-
-        <input class="color-class name edit-border" [(ngModel)]="organization.name" (blur)="name.valid && updateOrganizationName()"
-               name="name"
-               [style.color]="organization.color" i18n-title="@@header.organization.name"
-               title="Edit name of the organization"
-               #name="ngModel" required minlength="3" maxlength="24">
+<div class="col-12 col-lg-11 col-xl-10 m-auto" *ngIf="organization">
```
Comment (livthomas) [inline src/app/workspace/organization/organization-settings.component.html]: This component is just huge. We should try to split it into the smaller ones in the future.

File: src/app/core/dto/group.ts [hunk 5]
```diff
@@ -19,8 +19,7 @@
 
 export interface Group {
 
-  id?: string;
+  id: string;
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/dto/group.ts]: We create new group with `GroupModel` interface. This is dto class and it should be alwas defined.

File: src/app/shared/utils/validator.ts (hunk 2 above)
Comment (kubedo8) [reply src/app/shared/utils/validator.ts]: Okay

File: src/app/core/store/users/users.state.ts [hunk 6]
```diff
@@ -21,14 +21,38 @@ import {createEntityAdapter, EntityState} from '@ngrx/entity';
...
 import {AppState} from '../app.state';
 import {UserModel} from './user.model';
+import {selectOrganizationByWorkspace} from "../organizations/organizations.state";
+import {selectAllGroups} from "../groups/groups.state";
+import {UserFilters} from "./user.filters";
+import {GroupModel} from "../groups/group.model";
 
 export interface UsersState extends EntityState<UserModel> {
 
+  filter: string
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/store/users/users.state.ts]: When user select another tab and return to `users` tab, filter is removed. I think that this behaviour is not very user friendly. Do you have any other solution?

File: src/app/shared/input/filter-box/filter-box.component.ts (hunk 1 above)
Comment (kubedo8) [reply src/app/shared/input/filter-box/filter-box.component.ts]: I wanted to distinguish between `input` variables and `member` variables.

File: src/app/workspace/organization/organization-settings.component.html (hunk 4 above)
Comment (kubedo8) [reply src/app/workspace/organization/organization-settings.component.html]: Do you think that making `resource-header` is good idea? I was thinking about that, but there are some differences so I am not sure about that.. But of course I can split this component into the smaller ones..

File: src/app/core/store/organizations/organizations.effects.ts [hunk 7]
```diff
@@ -74,12 +77,26 @@ export class OrganizationsEffects {
...
+
+      const paramMap = RouteFinder.getFirstChildRouteWithParams(this.router.routerState.root.snapshot).paramMap;
+      const orgCodeInRoute = paramMap.get('organizationCode');
+
+      if (orgCodeInRoute && orgCodeInRoute === oldOrganization.code && organization.code !== oldOrganization.code) {
+        const paths = this.router.routerState.snapshot.url.split("/").filter(path => path);
+        const index = paths.indexOf(oldOrganization.code, 1);
+        if(index !== -1){
+          paths[index] = organization.code;
+          actions.push(new RouterAction.Go({path: paths}))
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/store/organizations/organizations.effects.ts]: Okay

File: src/app/core/dto/group.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (livthomas) [reply src/app/core/dto/group.ts]: I mean when you create a new group object and send it to the server for the first time, it does not have `id` field defined, right?

File: src/app/core/store/users/users.state.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (livthomas) [reply src/app/core/store/users/users.state.ts]: I see your point. In that case I would recommend to pass such information using router's query params and remove it from them in `OnDestroy` method of the parent component.

File: src/app/workspace/organization/organization-settings.component.html (hunk 4 above)
Comment (livthomas) [reply src/app/workspace/organization/organization-settings.component.html]: I have not really analyzed it into the depth. If we create two components that are similar we can then work further and extract them into the new abstract one.

**Pull request merged**